Travis is missing doxygen warnings

https://travis-ci.org/sxs-collaboration/spectre/jobs/432531550  reports
```
Built target doc
/work/spectre/src/Utilities/Tuple.hpp:245: warning: Ignoring \brief command inside argument documentation
 ---> 7f659433f49c
```
but the build succeeds anyway.

--- a/.travis/BuildLinux.sh
+++ b/.travis/BuildLinux.sh
@@ -44,7 +44,7 @@ fi
 # can search the file for warnings
 if [ ${BUILD_DOC} ]; then
     make doc | tee doxygen.log
-    ! grep 'warning' ./doxygen.log
+    grep 'warning' ./doxygen.log && exit 1
 fi
 
 # Build the code and run tests

